datastream-kafka-connector/src/main/java/com/linkedin/datastream/connectors/kafka/AbstractKafkaConnector.java

```diff
@@ -102,6 +102,8 @@
   // multiple concurrent threads. If access is required to both maps then the order of synchronization must be
   // _runningTasks followed by _tasksToStop to prevent deadlocks.
   private final Map<DatastreamTask, ConnectorTaskEntry> _runningTasks = new HashMap<>();
+
+  // _tasksToStop contains the tasks which are needed to be stopped across various assignment changes.
```

Comment:
There is more clarity on what this map is storing and how it is important.

It contains the list of tasks that are pending stop. The periodic scheduled health check call will try to stop these tasks again, with the assumption that that the task is stuck somewhere in stop path or the stop is not completed yet.

I feel it will be very useful to add this information in the comments and ensure the variable name is more meaningful for better code readability for other users.